### DISCUSSION TITLE
generating exact GraphQL operations with client-preset

### DISCUSSION THREAD

**Discussion by saihaj** [dotansimha/graphql-code-generator#8693] [Ideas]
Today we have a full generated schema and then have the operations which are being used. 
> We could remove a lot of noise if we rewrite the code generation part and only generate the exact graphql operation types instead of also generating the whole schema  types
from @n1ru4l

**Comment by christiansany** [comment]
HI @saihaj 
I really like this idea. The generated Schema types are in my opinion overhead and I can't think of a scenario where we would want to use those types anyway. Having those types available may actually harm you and may lead to overfetching.
We've encountered where a property of a Component FunctionComponent is actually tied to a Schema Type instead of the actual Data needed to render the Component.
Only having the Fragment & Query Types available of your actual operations solves that Problem.

**Comment by nikolay-emrikh-stenn** [comment]
Usage of generated types other than operations could be dangerous. We ran into the problem when we can't get rid of these generated types (other than operations) since we're using client preset, thus setting [onlyOperationTypes](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript#onlyoperationtypes) option to true has no effect

**Comment by ericfreese** [comment]
Could this be as simple as just adding `onlyOperationTypes` option to the list of options forwarded through the client preset?

https://github.com/dotansimha/graphql-code-generator/blob/1ac08cca7a25a47b0488601c2a733294065a08c3/packages/presets/client/src/index.ts#L119-L134

**Reply by setchy** [reply]
Curious if anyone has tried this ☝️
